sshcode.go

```diff
@@ -79,9 +79,17 @@ func sshCode(host, dir string, o options) error {
 
 	flog.Info("starting code-server...")
 
-	if o.localPort == "" {
-		o.localPort, err = randomPort()
+	var bindHost string
+	var bindPort string
+	bindHost, bindPort, err = net.SplitHostPort(o.bindAddr)
+
+	if err != nil {
+		return xerrors.Errorf("failed to parse bind address: %w", err)
+	}
+	if bindPort == "" {
```

Comment:
I would probably pull this out into a separate function `func parseBindAddr(bindAddr string) (bindAddr string, err error)` that splits the address and ensures that neither host nor port is empty. If one is, it gets assigned a default value, `127.0.0.1` for host, and a random port for port and then join it back together with `net.JoinHostPort`